sendgrid/sendgrid.py

```diff
@@ -58,3 +57,14 @@ def api_key(self, value):
     @property
     def impersonate_subuser(self):
         return self._impersonate_subuser
+
+
+class SendGridClient(SendGridAPIClient):
+    def __init__(self, **opts):
+        super().__init__(**opts)
+
+    def send_mail(self, msg):
+        try:
+            return self.client.mail.send.post(request_body=msg.get())
+        except Exception as e:

```

Comment:
You are right. Your implementation looks more explicit. Like I have mentioned in the issue description, this is just an example of how `SendGridException` can be used. This is in a very nascent stage.
 Right now, catching an `Exception`, though doesn't makes as much sense as catching specific exceptions, we can still extract a lot of information about the error from it. Using the `__class__` attribute , we can determine the kind of error we have caught and process it accordingly. 
Of course, we can declare custom exceptions classes `SendGridEmptySubject` or `SendGridEmptyContent` for specific errors , but, in my opinion, this will become too verbose. 
Imagine a application that uses this module. Do you think it should care about adding except blocks for handling so many specific exceptions? Like `urllib` raises only HTTPError for all erroneous status code and not specific error for each of them. So we handle only HTTPError and process the error according to the status codes. (This is just my view. I **do not** strongly object to adding support for specific exceptions.) . Also the SendGrid v3 API returns good information about the errors.